printbuf.c

```diff
@@ -142,14 +177,29 @@ int sprintbuf(struct printbuf *p, const char *msg, ...)
 
 void printbuf_reset(struct printbuf *p)
 {
-  p->buf[0] = '\0';
-  p->bpos = 0;
+  if (p != NULL)
+  {
+	  if ( (p->size > 0) &&
+		   (p->buf != NULL)
+		 )
+	  {
+		  p->buf[0] = '\0';
+	  }
+
+	  p->bpos = 0;
+  }
 }
 
 void printbuf_free(struct printbuf *p)
 {
   if(p) {
-    free(p->buf);
+
+	if (p->buf != NULL)
+		free(p->buf);
+
+    p->buf = NULL;
```

Comment:
As with your changes in json_object_generic_delete, setting these to NULL is pointless.